src/utils.rs

```diff
@@ -109,3 +119,19 @@ impl<'a, S: Writeable> Writeable for MutexGuard<'a, S> {
         S::write(&**self, writer)
     }
 }
+
+/// Returns true if the given network is valid for the given destination network.
+/// This is used to prevent users from sending funds to the wrong network.
+/// We can't just compare the network directly because signet and testnet
+/// have conflicting address prefixes.
+pub(crate) fn is_valid_network(my_network: Network, dest_network: Network) -> bool {
```

Comment:
Is it worth considering defining `eq` / `not eq` for these networks? Then we wouldn't have to change all the `==` / `!=` anywhere.

Reply:
I think this is probably a better approach, there may be scenarios where we want to check testnet != signet

Reply:
Gotcha. Plus idk if we can easily since we don't own those traits. Hate that rust rule.